client/src/unifyfs.c

```diff
@@ -2828,9 +2830,20 @@ int unifyfs_transfer_file(const char* src, const char* dst, int parallel)
     }
 
     if (parallel) {
-        return do_transfer_file_parallel(src_path, dst_path, &sb_src, dir);
+        local_return_val =
+	    do_transfer_file_parallel(src_path, dst_path, &sb_src, dir);
     } else {
-        return do_transfer_file_serial(src_path, dst_path, &sb_src, dir);
+        local_return_val =
+	    do_transfer_file_serial(src_path, dst_path, &sb_src, dir);
+    }
+
+    // We know here that one (but not both) of the constituent files
+    // is in the unify FS.  We just have to decide if the *destination* file is.
+    // If it is, then now that we've transferred it, we'll set it to be readable
+    // so that it will be laminated and will be readable by other processes.
+    if (unify_dst) {
+        chmod(dst_path, 0444);
```

Comment:
@craigsteffen It's not clear to me if this will go through our chmod wrapper or not. When you tested this, did you see the `LOGDBG("chmod: setting ...")` message from `int UNIFYFS_WRAP(chmod)(const char* path, mode_t mode)`?

Reply:
Also, instead of setting 0444, we should remove the write bits from the existing permissions.